### PR TITLE
RAB-1011: Add fake mailer service

### PR DESCRIPTION
**Pull request by StevenVAIDIE**
**Description (for Contributor and Core Developer)**

In this PR, 
I mocked the MailerService into acceptance test. By doing this we are now able to assert that email have been sent

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: tests/back/Platform/Acceptance/NotificationBundle/FakeService/FakeMailNotifier.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+namespace AkeneoTest\Platform\Acceptance\NotificationBundle\FakeService;
+
+use Akeneo\Platform\Bundle\NotificationBundle\Email\MailNotifierInterface;
+use Webmozart\Assert\Assert;
+
+class FakeMailNotifier implements MailNotifierInterface
+{
+    private array $emailSent = [];
+
+    public function notify(array $users, $subject, $txtBody, $htmlBody = null, array $options = [])
```
Comment (ghoussard) [inline tests/back/Platform/Acceptance/NotificationBundle/FakeService/FakeMailNotifier.php:19]:
```suggestion
    public function notify(array $users, $subject, $txtBody, $htmlBody = null, array $options = []): void
```

File: tests/back/Platform/Acceptance/NotificationBundle/FakeService/FakeMailNotifier.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    {
+        foreach ($users as $user) {
+            $this->emailSent[$user->getEmail()][] = $subject;
+        }
+    }
+
+    public function notifyByEmail(string $recipient, string $subject, string $txtBody, $htmlBody = null, array $options = [])
```
Comment (ghoussard) [inline tests/back/Platform/Acceptance/NotificationBundle/FakeService/FakeMailNotifier.php:26]:
```suggestion
    public function notifyByEmail(string $recipient, string $subject, string $txtBody, $htmlBody = null, array $options = []): void
```

File: tests/back/Platform/Acceptance/NotificationBundle/FakeService/FakeMailNotifier.php [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    {
+        $this->emailSent[$recipient][] = $subject;
+    }
+
+    public function assertEmailHaveBeenSent(string $addressMail, string $subject)
```
Comment (ghoussard) [inline tests/back/Platform/Acceptance/NotificationBundle/FakeService/FakeMailNotifier.php:31]:
```suggestion
    public function assertEmailHaveBeenSent(string $addressMail, string $subject): void
```

File: tests/back/Platform/Acceptance/NotificationBundle/FakeService/FakeMailNotifier.php [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+            sprintf(
+                'No email have been sent to "%s" with subject "%s". Got "%s"',
+                $addressMail,
+                $subject,
+                implode(', ', $emailSubjectsSent)
+            ),
+        );
+    }
+
+    public function assertNoEmailHaveBeenSent()
```
Comment (ghoussard) [inline tests/back/Platform/Acceptance/NotificationBundle/FakeService/FakeMailNotifier.php:49]:
```suggestion
    public function assertNoEmailHaveBeenSent(): void
```

**Review by ghoussard — APPROVED** [review]

File: tests/back/Platform/Acceptance/NotificationBundle/FakeService/FakeMailNotifier.php (hunk 1 above)
Comment (grena) [inline tests/back/Platform/Acceptance/NotificationBundle/FakeService/FakeMailNotifier.php:19]: Maybe a phpdoc to know what object we'll have in `$users`?

File: tests/back/Platform/Acceptance/NotificationBundle/FakeService/FakeMailNotifier.php (hunk 3 above)
Comment (grena) [reply tests/back/Platform/Acceptance/NotificationBundle/FakeService/FakeMailNotifier.php:31]:
```suggestion
    public function assertEmailHasBeenSent(string $addressMail, string $subject): void
```

File: tests/back/Platform/Acceptance/NotificationBundle/FakeService/FakeMailNotifier.php [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+    public function notifyByEmail(string $recipient, string $subject, string $txtBody, $htmlBody = null, array $options = [])
+    {
+        $this->emailSent[$recipient][] = $subject;
+    }
+
+    public function assertEmailHaveBeenSent(string $addressMail, string $subject)
+    {
+        Assert::notEmpty($this->emailSent, 'No email have been sent');
+        Assert::keyExists($this->emailSent, $addressMail, 'No email have been sent to "%s"');
```
Comment (grena) [inline tests/back/Platform/Acceptance/NotificationBundle/FakeService/FakeMailNotifier.php:34]: > has been

**Review by grena — APPROVED** [review]

File: tests/back/Platform/Acceptance/NotificationBundle/FakeService/FakeMailNotifier.php (hunk 5 above)
Comment (StevenVAIDIE) [reply tests/back/Platform/Acceptance/NotificationBundle/FakeService/FakeMailNotifier.php:34]: C'est toi qui est has been

**Review by pierrejolly — APPROVED** [review]

**Pull request merged**